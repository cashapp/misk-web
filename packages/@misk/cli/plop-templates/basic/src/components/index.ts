import HowToComponent from "./HowToComponent"
export { HowToComponent }
