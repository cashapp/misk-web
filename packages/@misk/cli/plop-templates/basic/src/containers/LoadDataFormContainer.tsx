import { Button, FormGroup, Intent, InputGroup } from "@blueprintjs/core"
import {
  handler,
  mergeSagaMapKeysToTags,
  simpleSelectorGet
} from "@misk/simpleredux"
import * as React from "react"
import { connect } from "react-redux"
import {
  IDispatchProps,
  IState,
  mapDispatchToProps,
  mapStateToProps
} from "src/ducks"

interface OwnProps {
  tag: string
}

export const LoadDataFormContainer = (
  props: IState & IDispatchProps & OwnProps
) => {
  const { tag } = props
  const seedData = {
    key: "cars",
    range: [50, 75],
    url: "https://cashapp.github.io/misk-web/examples/data/demo/cars.json"
  }
  const keyTagLookup = {
    key: `${tag}::dataKey`,
    range: `${tag}::dataRange`,
    url: `${tag}::dataUrl`
  }
  return (
    <div>
      <FormGroup>
        <Button
          onClick={handler.simpleMergeData(props, tag, {
            mergeSaga: mergeSagaMapKeysToTags(props, "data", keyTagLookup),
            overrideArgs: seedData
          })}
          text={"Use Example Values"}
        />
        <InputGroup
          placeholder={
            "data endpoint URL (example: https://cashapp.github.io/misk-web/examples/data/demo/cars.json"
          }
          onChange={handler.simpleMergeData(props, `${tag}::dataUrl`)}
          value={simpleSelectorGet(
            props.simpleRedux,
            [`${tag}::dataUrl`, "data"],
            ""
          )}
          type={"url"}
        />
        <InputGroup
          placeholder={"network response data object key (example: cars)"}
          onChange={handler.simpleMergeData(props, `${tag}::dataKey`)}
          value={simpleSelectorGet(
            props.simpleRedux,
            [`${tag}::dataKey`, "data"],
            ""
          )}
        />
        <Button
          onClick={handler.simpleHttpGet(
            props,
            `${tag}::dataRequest`,
            simpleSelectorGet(props.simpleRedux, [`${tag}::dataUrl`, "data"])
          )}
          intent={Intent.PRIMARY}
          loading={simpleSelectorGet(props.simpleRedux, [
            `${tag}::dataRequest`,
            "loading"
          ])}
          text={"GET Data"}
        />
      </FormGroup>
    </div>
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoadDataFormContainer)
