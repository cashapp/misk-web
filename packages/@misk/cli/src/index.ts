#!/usr/bin/env node

import yargs from "yargs"

yargs
  .scriptName("miskweb")
  .commandDir("commands")
  .demandCommand()
  .option("e", {
    alias: "each",
    describe: "run command in all subdirectories that have miskTab.json",
    type: "boolean"
  })
  .help()
  .version()
  .usage(
    "Misk Web CLI Usage: $ $0 <command> <required arg> [optional arg] [options]"
  )
  .strict().argv
