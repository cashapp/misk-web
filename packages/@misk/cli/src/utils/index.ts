import * as fs from "fs-extra"
import { cd, exec, exit, pwd } from "shelljs"

export * from "./generate"
export * from "./handleCommand"
export * from "./miskTabUtilities"
export * from "./resolveNpmVersion"

export enum MiskPkg {
  "cli" = "@misk/cli",
  "common" = "@misk/common",
  "core" = "@misk/core",
  "dev" = "@misk/dev",
  "prettier" = "@misk/prettier",
  "simpleredux" = "@misk/simpleredux",
  "test" = "@misk/test",
  "tslint" = "@misk/tslint"
}

export interface IMiskTabJSON {
  name: string // name of tab in Title Case
  output_path: string // output path for Webpack build
  port: number // port for Webpack-Dev-Server
  rawGitginore: string // prebuild permanent add to .gitignore file
  rawIndex: boolean // ignore Webpack injection of source links to index.html
  rawPackageJson: any // prebuild permanent add/override to package.json file
  rawTsconfig: any // prebuild permanent add/override to tsconfig.json file
  rawTslint: any // prebuild permanent add/override to tslint.json file
  rawWebpackConfig: any // prebuild permanent add to webpack.config.js file
  relative_path_prefix: string // override default URL for tab: /_tab/{slug}/
  slug: string // unique slug used in URL path
  useWebpackBundleAnalyzer: boolean // turn off/on webpack bundle analyzer reports
  useWebpackExternals: boolean // turn off/on thin build by including externals in Webpack build
  version: string // Misk Web release version or keyword (alpha, latest)
  zipOnBuild: boolean // zip relevant source code of tab into {slug}.tgz after each build
  ___DeprecatedKeys: string // divider to the miskTab.json to improve maintainability
}

export enum Files {
  "gitignore" = ".gitignore",
  "masterDependencies" = "master-dependencies.json",
  "miskTab" = "miskTab.json",
  "old" = ".old_build_files",
  "package" = "package.json",
  "packageLock" = "package-lock.json",
  "prettier" = "prettier.config.js",
  "setupMiskTest" = "setupMiskTest.js",
  "tsconfig" = "tsconfig.json",
  "tslint" = "tslint.json",
  "webpack" = "webpack.config.js",
  "yarnLock" = "yarn.lock"
}

export const JsonOptions = { spaces: 2 }

export const logFormatter = (
  tag: string,
  msg?: string,
  dir: string = pwd().stdout
) =>
  `[${tag.toUpperCase()}][${dir
    .split("/")
    .pop()
    .toUpperCase()}] ${msg}`

export const logDebug = (
  tag: string,
  msg?: string,
  dir: string = pwd().stdout
) => console.log(logFormatter(tag, msg, dir))

export const makePath = (...segments: string[]) => `${segments.join("/")}`

export const parseArgs = (...args: any): { dir: string; rawArgs: any } => ({
  dir: pwd().stdout,
  rawArgs: args
})

export const remove = async (path: string) => {
  try {
    fs.remove(path)
  } catch (e) {
    console.log(`[ERROR] ${e}`)
  }
}

export const execute = (cmd: string, ...args: any) => {
  const { dir } = parseArgs(...args)
  cd(dir)
  const terminal = exec(cmd)
  terminal.stdout
  if (terminal.code) {
    console.log(`\n[ERROR][CODE] Shell command exited with code ${terminal.code}:\n\`${cmd}\``)
    exit(terminal.code)
  }
}

export const npmRunScript = (cmd: string, prebuild: boolean = false) =>
  `${prebuild ? "miskweb prebuild && " : ""}npm run-script ${cmd}`
