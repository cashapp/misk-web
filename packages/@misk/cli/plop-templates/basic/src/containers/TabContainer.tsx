import * as React from "react"
import { connect } from "react-redux"
import { HowToComponent } from "src/components"
import { LoadDataFormContainer, LoadDataTableContainer } from "src/containers"
import {
  IDispatchProps,
  IState,
  mapDispatchToProps,
  mapStateToProps
} from "src/ducks"

const TabContainer = (props: IState & IDispatchProps) => {
  const tag = "LoadData"
  return (
    <div>
      <HowToComponent />
      <LoadDataFormContainer tag={tag} />
      <LoadDataTableContainer tag={tag} />
    </div>
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(TabContainer)
