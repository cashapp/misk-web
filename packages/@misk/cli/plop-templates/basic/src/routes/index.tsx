import * as React from "react"
import { Route, Switch } from "react-router"
import { TabContainer } from "src/containers"

const routes = (
  <div>
    <Switch>
      <Route path="/_admin/{{name}}/" component={TabContainer} />
      {/* Do not include a Route without a path or it will display during on all tabs: */}
      {/* <Route component={TabContainer} /> */}
    </Switch>
  </div>
)

export default routes
