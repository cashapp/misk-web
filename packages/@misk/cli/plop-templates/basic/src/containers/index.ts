import LoadDataFormContainer from "./LoadDataFormContainer"
import LoadDataTableContainer from "./LoadDataTableContainer"
import TabContainer from "./TabContainer"
export { LoadDataFormContainer, LoadDataTableContainer, TabContainer }
