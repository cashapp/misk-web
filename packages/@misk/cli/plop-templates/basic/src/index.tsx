import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./components"
export * from "./containers"

createIndex("{{name}}", createApp(routes), Ducks)
