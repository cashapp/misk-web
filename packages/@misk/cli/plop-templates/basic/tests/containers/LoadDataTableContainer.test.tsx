import React from "react"
import { cleanup } from "@testing-library/react"
import { LoadDataTableContainer } from "src/containers"
import { renderWithRedux } from "../upstreamableTestUtilities"
import { stateWithData } from "./testFixtures"

describe("LoadDataTableContainer", () => {
  afterEach(cleanup)
  it("LoadDataTableContainer can render with redux", () => {
    const { asFragment } = renderWithRedux(
      <LoadDataTableContainer tag={"LoadData"} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("LoadDataTableContainer can render with data", () => {
    const { asFragment } = renderWithRedux(
      <LoadDataTableContainer tag={"LoadData"} />,
      stateWithData
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
