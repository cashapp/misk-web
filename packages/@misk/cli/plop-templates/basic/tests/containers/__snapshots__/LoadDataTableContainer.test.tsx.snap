// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadDataTableContainer LoadDataTableContainer can render with data 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bp3-progress-bar bp3-no-animation"
    >
      <div
        class="bp3-progress-meter"
      />
    </div>
    <table
      class="bp3-html-table bp3-html-table-bordered bp3-html-table-striped"
    >
      <thead>
        <tr>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`LoadDataTableContainer LoadDataTableContainer can render with redux 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bp3-progress-bar bp3-no-animation"
    >
      <div
        class="bp3-progress-meter"
      />
    </div>
    <table
      class="bp3-html-table bp3-html-table-bordered bp3-html-table-striped"
    >
      <thead>
        <tr>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
          <th>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
          <td>
            <p
              class="bp3-skeleton"
            >
              lorem ipsum 1234 5678
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
